x/axelarnet/keeper/msg_server.go

```diff
@@ -410,6 +412,81 @@ func (s msgServer) RetryIBCTransfer(c context.Context, req *types.RetryIBCTransf
 	return &types.RetryIBCTransferResponse{}, nil
 }
 
+func (s msgServer) ExecuteMessage(c context.Context, req *types.ExecuteMessageRequest) (*types.ExecuteMessageResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	chain, ok := s.nexus.GetChain(ctx, req.Chain)
+	if !ok {
+		return nil, fmt.Errorf("invalid chain %s", req.Chain)
+	}
+
+	if !s.nexus.IsChainActivated(ctx, chain) {
+		return nil, fmt.Errorf("chain %s is not activated", chain.Name)
+	}
+
+	generalMessage, ok := s.nexus.GetMessage(ctx, nexus.MessageID{ID: req.ID, Chain: req.Chain})
+	if !ok {
+		return nil, fmt.Errorf("message %s not found", req.ID)
+	}
+
+	if !s.nexus.IsAssetRegistered(ctx, chain, generalMessage.Asset.GetDenom()) {
+		return nil, fmt.Errorf("asset %s is not registered on chain %s", generalMessage.Asset.GetDenom(), req.Chain)
+	}
+
+	if !generalMessage.Match(req.Payload) {
+		return nil, fmt.Errorf("payload hash does not match")
+	}
+
+	if !generalMessage.Is(nexus.Approved) {
+		return nil, fmt.Errorf("general message %s already executed", req.ID)
+	}
+
+	version, payload, err := types.UnpackPayload(req.Payload)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "invalid payload with version")
+	}
+
+	memo, err := constructMessage(generalMessage, version, payload)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "invalid payload")
```

Comment:
didn't we already validated the payload in the previous call?

Reply:
we are not validating here, it tries to decode the abi encoded payload  and converts to a native/wasm message